src/uct/ib/base/ib_iface.c

```diff
@@ -569,8 +576,33 @@ ucs_status_t uct_ib_iface_get_device_address(uct_iface_h tl_iface,
     return UCS_OK;
 }
 
+static int
+uct_ib_iface_roce_is_reachable_via_routing(const uct_ib_device_gid_info_t *local_gid_info,
+                                           const uct_ib_address_t *remote_ib_addr,
+                                           unsigned prefix_bits)
+{
+    sa_family_t addr_family;
+    size_t addr_offset;
+    uint8_t *local_addr;
+    uint8_t *remote_addr;
+
+    if (local_gid_info->roce_info.ver != UCT_IB_DEVICE_ROCE_V2) {
+        return 1; /* We assume it is, but actually there's no good test */
+    }
+
+    addr_family = local_gid_info->roce_info.addr_family;
+    addr_offset = uct_ib_gid_offset_of_roce_v2_addr(addr_family);
+    local_addr  = UCS_PTR_BYTE_OFFSET(&local_gid_info->gid, addr_offset);
+    remote_addr = UCS_PTR_BYTE_OFFSET(&remote_ib_addr->flags + 1, addr_offset);
+
+    ucs_assert((prefix_bits / 8) <= addr_offset);
+
+    return ucs_bitwise_is_equal(local_addr, remote_addr, prefix_bits);
```

Comment:
pls add use_debug with details if the result is false (see other places in uct_ib_iface_roce_is_reachable as example)